hal/src/sercom/v2/dma.rs

```diff
@@ -11,12 +11,151 @@ use crate::{
         Beat, Buffer, Transfer, TriggerAction,
     },
     sercom::v2::{
+        i2c::{self, I2c},
         spi::{self, Spi},
         uart::{self, Uart},
         Sercom,
     },
 };
 
+//=============================================================================
+// I2C DMA transfers
+//=============================================================================
+
+/// Token type representing an [`I2c`](super::i2c::I2c) for which the bus is
+/// ready to start a transaction. For use with
+/// [`send_with_dma`](super::i2c::I2c::send_with_dma) and
+/// [`receive_with_dma`](super::i2c::I2c::send_with_dma).
+pub struct I2cBusReady;
+
+unsafe impl<C: i2c::AnyConfig> Buffer for I2c<C> {
+    type Beat = i2c::Word;
+
+    #[inline]
+    fn dma_ptr(&mut self) -> *mut Self::Beat {
+        self.data_ptr()
+    }
+
+    #[inline]
+    fn incrementing(&self) -> bool {
+        false
+    }
+
+    #[inline]
+    fn buffer_len(&self) -> usize {
+        1
+    }
+}
+
+impl<C: i2c::AnyConfig> I2c<C> {
+    /// Initialize the bus to start receiving with DMA.
+    ///
+    /// In reality, this function only checks whether or not the I2C bus is
+    /// ready to accept a new transaction. A call to [`init_dma_transfer`]
+    /// should immediately be followed by a call to [`send_with_dma`] or
+    /// [`receive_with_dma`].
+    ///
+    /// ```
```

Comment:
Does this need a `no_run` comment? Compile only doc test

Reply:
I added `no_run` pretty much everywhere, but to be honest our doctests pretty much all fail. This could be an action item to fix, but to be fair the setup required to make every example snippet compile is pretty large.

Reply:
yeah, that would be a large undertaking